lib/Elastica/Pipeline.php

```
@@ -0,0 +1,220 @@
+
+    /**
+     * Create a new Pipeline Object.
+     *
+     * @param Client $client
+     *
+     */
+    public function __construct(Client $client)
+    {
+        $this->_client = $client;
+    }
+
+    /**
+     * Create a Pipeline.
+     *
+     * @return Response
+     * @link https://www.elastic.co/guide/en/elasticsearch/reference/current/put-pipeline-api.html
+     */
+    public function create()
+    {
+        if (empty($this->id)) {
+            throw new InvalidException('You should set a valid pipeline id');
+        }
+
+        if (empty($this->_params['description'])) {
+            throw new InvalidException('You should set a valid processor description.');
+        }
+
+        if (empty($this->_processors['processors'])) {
+            throw new InvalidException('You should set a valid processor of type Elastica\Processor\AbstractProcessor.');
+        }
+
+        $endpoint = new Put();
+        $endpoint->setID($this->id);
+        $endpoint->setBody($this->toArray());
+
+        return $this->requestEndpoint($endpoint);
+    }
+
+    /**
+     * Get a Pipeline Object.
+     * @param string $id Pipeline name
+     * @return Response
+     *
+     * @link https://www.elastic.co/guide/en/elasticsearch/reference/current/get-pipeline-api.html
+     */
+    public function getPipeline(string $id)
+    {
+        $endpoint = new Get();
+        $endpoint->setID($id);
+
+        return $this->requestEndpoint($endpoint);
+    }
+
+    /**
+     * Delete a Pipeline.
+     *
+     * @param string $id Pipeline name
+     * @return Response
+     *
+     * @link https://www.elastic.co/guide/en/elasticsearch/reference/current/delete-pipeline-api.html
+     */
+    public function deletePipeline(string $id)
+    {
+        $endpoint = new Delete();
+        $endpoint->setID($id);
+
+        return $this->requestEndpoint($endpoint);
+    }
+
+    /**
+     * @todo implement simulate API
+     */
+    public function simulate()
+    {
+
```

Should we throw an exception for now?

---

ok I raise a *NotImplementedException('simulate API on Pipeline not yet implemented.');*